[ML] Add new include flag to GET inference/<model_id> API for model training metadata (#61922)
elastic/elasticsearch: docs/reference/ml/df-analytics/apis/get-inference-trained-model.asciidoc

and then added to another cluster. Default is false.

`from`::
(Optional, integer)
include::{es-repo-dir}/ml/ml-shared.asciidoc[tag=from-models]

`include`::
(Optional, string)
A comma delimited string of optional fields to include in the response body.
Valid options are:

>  - definition: to include the model definition

Reviewer: ```suggestion
 - `definition`: Includes the model definition
```

>  - total_feature_importance: to include the total feature importance for the

Reviewer: ```suggestion
 - `total_feature_importance`: Includes the total feature importance for the
```

>    training feature sets. This field will be available in the `metadata` field.

Reviewer: ```suggestion
   training data sets. This field is available in the `metadata` field in the response body.
```

`metadata`:::
(object)
An object containing metadata about the trained model. For example, models
created by {dfanalytics} contain `analysis_config` and `input` objects.
.Properties of metadata
[%collapsible%open]
=====
`total_feature_importance`:::
(array)

> An array of the total feature importance for each training feature used from

Reviewer: > An array of the total feature importance for each training feature used from

AFAIK we've never defined a "training feature".  Can we just say feature? If not, this term needs to be explained.

`metadata`:::
(object)
An object containing metadata about the trained model. For example, models
created by {dfanalytics} contain `analysis_config` and `input` objects.

> .Properties of metadata

Reviewer: This isn't formatting properly, so I think a continuation character is required.
```suggestion
+
.Properties of metadata
```

Reviewer: I'll fix this in https://github.com/elastic/elasticsearch/pull/62643
